### PR TITLE
YLP-1981 Some data is uploaded with wrong time information

### PR DESCRIPTION
**Pull request by yanngit**

### CODE REVIEW

File: data/store/mongo/mongo_bucket_store_client_test.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+			want: &schema.Metadata{
+				Id:                  "metadata1234",
+				CreationTimestamp:   testTime,
+				UserId:              "123456789",
+				OldestDataTimestamp: testTime,
+				NewestDataTimestamp: testTime,
+			},
+		},
+		{
+			name: "given a normal data timestamp should update oldest metadata",
```
Comment (CedricPerotto) [inline data/store/mongo/mongo_bucket_store_client_test.go:69]: We could add a test that updates the NewestDataTimestamp only.

File: data/store/mongo/mongo_bucket_store_client.go [hunk 2]
```diff
@@ -243,7 +243,10 @@ func (c *MongoBucketStoreClient) BuildUserMetadata(incomingUserMetadata *schema.
 			NewestDataTimestamp: dataTimestamp,
 		}
 	} else {
-		if incomingUserMetadata.OldestDataTimestamp.After(dataTimestamp) {
+		//Linked to YLP-1981, in some situation the DBLG1 is sending a data with a timestamp in the near 1970's ...
+		//We do not want to update our metadata with this value. The CBG will be recorded with the 1970's date to keep
+		//a trace of it, but it won't be displayed since the data is erroneous.
+		if incomingUserMetadata.OldestDataTimestamp.After(dataTimestamp) && dataTimestamp.Year() > 2015 {
```
Comment (CedricPerotto) [inline data/store/mongo/mongo_bucket_store_client.go]: This 2015 year could be an env parameter (if we have to be interoperable we could have some data before 2015...)

**Review by CedricPerotto — COMMENTED** [review]
LGTM just a few improvements for tests & make this year parameter configurable).

File: data/store/mongo/mongo_bucket_store_client_test.go (hunk 1 above)
Comment (yanngit) [reply data/store/mongo/mongo_bucket_store_client_test.go:69]: done

File: data/store/mongo/mongo_bucket_store_client.go (hunk 2 above)
Comment (yanngit) [reply data/store/mongo/mongo_bucket_store_client.go]: done

**Review by CedricPerotto — APPROVED** [review]

**Pull request merged**